pytket/extensions/aqt/multi_zone_architecture/circuit/multizone_circuit.py

```diff
@@ -316,15 +319,19 @@ def move_qubit(self, qubit: int, new_zone: int) -> None:
             else:
                 position_in_zone = VirtualZonePosition.VirtualLeft
 
-        self.pytket_circuit.add_custom_gate(move_gate, [new_zone], [qubit])
-        self.add_move_barrier()
+        if not precompiled:
+            self.pytket_circuit.add_custom_gate(move_gate, [new_zone], [qubit])
+            self.add_move_barrier()
         old_zone_qubits.remove(qubit)
         if position_in_zone is VirtualZonePosition.VirtualLeft:
             self.zone_to_qubits[new_zone].insert(0, qubit)
         else:
             self.zone_to_qubits[new_zone].append(qubit)
         self.qubit_to_zones[qubit].append(new_zone)
         self.multi_zone_operations[qubit].append(move_operations)
+        if precompiled:
```

Comment:
is it checked somewhere if the pre compilation is correct?

Reply:
In this code, precompiled is just a switch to either add the physical operations directly (because pytket compilation is complete and we only need to do routing) or not, i.e. mark where they go and add later after compilation. (because the circuit compilation would remove them). precompiled=True should normally never be needed for a user of the library, it's only used for internal automatic routing.

I've added more info in the docstring to make this clearer